### PR TITLE
fix(plasma-b2c): Avoid text overflow for Textarea 

### PR DESCRIPTION
**Pull request by Yakutoc**
Проблема: 

Текст наезжал на нижний блок с подсказками.   

Было: 
Снимок экрана 2023-02-07 в 12 18 21

Стало:
Снимок экрана 2023-02-07 в 12 17 47
 

:baby_chick: Download canary assets:

[colors--canary.311.4142673286.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.311.4142673286.xml)  
[color_metro_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.311.4142673286.swift)  
[color_metro_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.311.4142673286.kt)  
[color_metro_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.311.4142673286.ts)  
[color_sbermarket_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.311.4142673286.swift)  
[color_sbermarket_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.311.4142673286.kt)  
[color_sbermarket_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.311.4142673286.ts)  
[color_sberprime_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.311.4142673286.swift)  
[color_sberprime_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.311.4142673286.kt)  
[color_sberprime_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.311.4142673286.ts)  
[color_selgros_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.311.4142673286.swift)  
[color_selgros_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.311.4142673286.kt)  
[color_selgros_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.311.4142673286.ts)  
[color_smbusiness_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.311.4142673286.swift)  
[color_smbusiness_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.311.4142673286.kt)  
[color_smbusiness_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.311.4142673286.ts)  
[PlasmaTokensColor--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.311.4142673286.swift)  
[shadow_sbermarket_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.311.4142673286.ts)  
[typo_mage_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.311.4142673286.swift)  
[typo_mage_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.311.4142673286.kt)  
[typo_mage_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.311.4142673286.ts)  
[typo_plasma_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.311.4142673286.swift)  
[typo_plasma_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.311.4142673286.kt)  
[typo_plasma_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.311.4142673286.ts)  
[typo_ruler_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.311.4142673286.swift)  
[typo_ruler_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.311.4142673286.kt)  
[typo_ruler_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.311.4142673286.ts)  
[typo_sage_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.311.4142673286.swift)  
[typo_sage_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.311.4142673286.kt)  
[typo_sage_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.311.4142673286.ts)  
[typo_sbermarket_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.311.4142673286.swift)  
[typo_sbermarket_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.311.4142673286.kt)  
[typo_sbermarket_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.311.4142673286.ts)  
[typo_soulmate_ios-swift--canary.311.4142673286.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.311.4142673286.swift)  
[typo_soulmate_kotlin--canary.311.4142673286.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.311.4142673286.kt)  
[typo_soulmate_react-native--canary.311.4142673286.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.311.4142673286.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.137.1-canary.311.4142673286.0
  # or 
  yarn add @salutejs/plasma-b2c@1.137.1-canary.311.4142673286.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-311/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-311/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-311/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-311/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-311/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-311/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-311/temple-storybook/

**Review by Yeti-or — COMMENTED** [review]
В size_xs хелперы почему-то изменили положение, это норм?

**Comment by Yakutoc** [comment]
@Yeti-or 

> В size_xs хелперы почему-то изменили положение, это норм?

Да, это нормально, потому что меняется `padding-bottom: 12 => 9px`

Я еще отдам @gibushnev на проверку.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-311/

**Comment by Yakutoc** [comment]
@Yeti-or После ревью, внес изменения для padding-right: у самого textarea, блока с подсказками и иконки.

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-311/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-311/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-311/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-311/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-311/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-311/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-311/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-311/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-311/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-311/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-311/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-311/temple-storybook/

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -137,11 +154,39 @@ export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
...
+        const handleFocus = useCallback(
+            (event: FocusEvent<HTMLTextAreaElement>) => {
+                if (!onFocus) {
+                    return;
+                }
+
+                onFocus(event);
+
+                setFocusState(true);
+            },
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:166]: А для чего это понадобилось?

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:166]: @neretin-trike Чтобы применить стили фокуса и для блока с подсказками, они (подсказки) теперь в не самой textarea

Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:166]: Понял, а с помощью css селекторов это разрулить не выйдет? Типа проверить, что у соседнего (выложенного) элемента висит псевдо класс :focus и навесить стиль в зависимости от этого

Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:166]: Или, как в нашем случае скорее всего, проверить наличие атрибута focus_add_чототам (не помню точное название, когда фокусом на элемент встаёшь он добавляется)

Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:166]: @neretin-trike Я проверю, но кажется можно так сделать. Хорошая идея! Спасибо

**Review by gibushnev — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-311/

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx]: Итог: Да можно отказаться от решения на js. Переделал на css.

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-311/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-311/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-311/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-311/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-311/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-311/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 2]
```diff
@@ -29,14 +31,22 @@ export interface TextAreaProps extends Omit<BaseProps, 'size'> {
     size?: 'l' | 'm' | 's' | 'xs';
 }
 
-const StyledTextArea = styled(BaseArea)`
+const StyledTextArea = styled(BaseArea)<{ $isHelper: boolean }>`
     ${applyInputStyles};
 
+    --computed-height: calc(9.375rem - var(--field-helpers-padding-top) - var(--padding-bottom) - 14px);
+
     padding: var(--padding);
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:39]: а нам не хочется padding разбить уже на bottom / left / rigth ?

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -78,15 +102,10 @@ const TextAreaWrapper = styled(TextFieldRoot)<{ $properties: ReadonlyArray<Simpl
...
-`;
-
-const StyledFieldHelpers = styled(FieldHelpers)<{ width: number }>`
-    bottom: 0.75rem;
 
-    padding-left: var(--padding-x);
-    padding-right: var(--padding-x);
-
-    width: ${({ width }) => width}px;
+    textarea[data-focus-visible-added] + ${StyledFieldHelpers} {
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextArea/TextArea.tsx:106]: а это чисто текстарии аттрибут?

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextArea/TextArea.tsx (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/TextArea/TextArea.tsx:106]: @Yeti-or Да, в этом случае да. Объект фокуса `textarea`.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-311/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-311/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-311/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-311/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-311/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-311/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-311/temple-storybook/

**Pull request merged**